R/teal.osprey.R

```diff
@@ -8,3 +8,10 @@
 #' @keywords internal
 #'
 NULL
+
+# Needed to avoid R CMD note on no visible binding
+utils::globalVariables(c(
```

Comment:
no - please don't
use `.$data` instead

Reply:
reference: https://dplyr.tidyverse.org/articles/in-packages.html?q=R%20CMD%20CHECK#data-masking-and-tidy-selection-notes

the usage of `utils::globalVariables` is too powerful and has side effects of hiding meaningful errors

Reply:
It makes more sense to me to allow this NOTE to pass.

Reply:
If we wanted to do this correctly then yes - that would be the desired way to fix it. That's why we have a separate task for it. If we go your way (which is much faster I know) then we would have to have yet another task to convert it into `.data$` - then what's the difference to the current one? :)

I hope that this is already covered in other, actively developed packages.

Reply:
I'll try to edit the dplyr code, but if it causes the headache with NOTES maybe we should get rid of dplyr at all, utiliizing the benefit of no tidyverse dependencies, hence no possible breaking changes in the future? 

teal.osprey has around ~30 dplyr calls, but teal.goshawk has +120 dplyr calls - I wonder what's happening in there.

Reply:
:popcorn:

Reply:
Hey @pawelru I think there is a bit more besides NSE of dplyr in here. 
Can you maybe have a look at `conmed_level` and `conmed_var` variables in `srv_g_heatmap_bygrade` inside `R/tm_g_heat_bygrade.R` and propose how to fix this?

```{R}
  srv_g_heatmap_bygrade : <anonymous>: no visible binding for global
    variable 'conmed_level'
  srv_g_heatmap_bygrade : <anonymous>: no visible binding for global
    variable 'conmed_var'
```

I think it's related to the substitution, where we later use those variables in `expr` statement
![image](https://github.com/insightsengineering/teal.osprey/assets/133694481/d28d4688-9512-4bb5-bc07-623b11fbcc3f)

Reply:
in private conversation pawelru recommended to change the order of expr and env which solves this particular issue 
![image](https://github.com/insightsengineering/teal.osprey/assets/133694481/488bb380-d55c-47b5-9c13-81adaca7560a)

Reply:
looks like it solved most of the issues

Reply:
I'll try to see if we can live without .data$ now